crates/matrix-sdk/src/event_cache/paginator.rs

```diff
@@ -219,11 +219,19 @@ impl<PR: PaginableRoom> Paginator<PR> {
     /// (running /context or /messages).
     pub(super) fn set_idle_state(
         &self,
+        next_state: PaginatorState,
```

Comment:
Thanks for the review!

Small API question here: I could've kept the previous code, *but*! if I did that, there would not be any state notification, because the code below uses `self.state.set_if_not_eq(Idle)`, and the previous value was still `Idle`. Using `Initial` is fine because it's also the initial set you get when creating an empty paginator. Another way to fix that would've been to replace `set_if_not_eq` with `set`, which forces a notification (and maybe check in the timeline that we're not deduplicating such updates. Any preference over here?

Reply:
I think using `Initial` makes sense and as you said prevents the clients from messing up the integration by deduplicating the state updates, which is something we tend to do on other places.

Reply:
Good point, thanks!